### PR TITLE
ping.ext_host: rewrite to support passt tests

### PR DESCRIPTION
**Pull request by yanglei-rh**
Rewrite the code to support passt tests

ID:2619
Signed-off-by: Lei Yang leiyang@redhat.com

### CODE REVIEW

**Comment by yanglei-rh** [comment]
Hi @PaulYuuu Could you please help me review this patch? Thanks in advance.

**Comment by yanglei-rh** [comment]
Hi @fbq815 @zhenyzha Could you please help review this patch to confirm this change doesn't influence other platform? Many thanks.

**Comment by zhenyzha** [comment]
LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

` (1/1) Host_RHEL.m9.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.aarch64.page_4k.io-github-autotest-qemu.virtual_nic.with_stress.arm64-pci: PASS (1574.70 s)`

**Review by zhenyzha — APPROVED** [review]

File: generic/tests/ping.py [hunk 1]
```diff
@@ -54,10 +54,11 @@ def run(test, params, env):
     ping_ext_host = params.get("ping_ext_host", "no") == "yes"
     pre_cmd = params.get("pre_cmd", None)
     vm = env.get_vm(params["main_vm"])
+    serial_status = params.get("serial_login", "no")
 
     error_context.context("Login to guest", test.log.info)
     vm.verify_alive()
-    session = vm.wait_for_login(timeout=timeout)
+    session = vm.wait_for_login(timeout=timeout, serial=serial_status)
```
Comment (PaulYuuu) [inline generic/tests/ping.py:61]: Please check the type of `serial` required.

File: generic/tests/ping.py [hunk 2]
```diff
@@ -108,14 +109,15 @@ def run(test, params, env):
                                   interval, session=session, count=counts)
 
         # ping with flood
-        if not ping_ext_host or params.get("os_type") == "linux":
-            error_context.context("Flood ping test", test.log.info)
-            _ping_with_params(test, params, ip, interface,
-                              session=session, flood=True)
+        if params.get("flood_ping", "yes") == "yes":
```
[the remark this answers is not in the document]
Comment (yanglei-rh) [reply generic/tests/ping.py:112]: I don't forgot it, just I thought it was only used to get `Ture/False`.

File: generic/tests/ping.py (hunk 1 above)
Comment (yanglei-rh) [reply generic/tests/ping.py:61]: Updated, please help review this patch again, thanks a lot.

**Comment by PaulYuuu** [comment]
```python
    def get_boolean(self, key, default=False):
        """
        Check if a config option is set to a default affirmation or not.

        :param key: config option key
        :type key: str
        :param bool default: whether to assume "yes" or "no" if param
                             does not exist (defaults to False/"no").
        :return whether option is set to 'yes'
        :rtype: bool
        """
        value = self.get(key, "yes" if default else "no")
        if value in ("yes", "on", "true"):
            return True
        if value in ("no", "off", "false"):
            return False
        raise ValueError("Cannot get boolean parameter value for %s: %s", key, value)
```

**Comment by yanglei-rh** [comment]
> ```python
>     def get_boolean(self, key, default=False):
>         """
>         Check if a config option is set to a default affirmation or not.
> 
>         :param key: config option key
>         :type key: str
>         :param bool default: whether to assume "yes" or "no" if param
>                              does not exist (defaults to False/"no").
>         :return whether option is set to 'yes'
>         :rtype: bool
>         """
>         value = self.get(key, "yes" if default else "no")
>         if value in ("yes", "on", "true"):
>             return True
>         if value in ("no", "off", "false"):
>             return False
>         raise ValueError("Cannot get boolean parameter value for %s: %s", key, value)
> ```

Thanks for help sync it.

**Review by fbq815 — APPROVED** [review]
LGTM
qemu.ping.ext_host.s390-virtio: PASS (692.72 s)

**Comment by yanglei-rh** [comment]
Hi @PaulYuuu Could you please help me review it again, thanks a lot. It can test pass on my side.

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**